src/common/lib/metamask/index.js

```diff
@@ -80,4 +81,32 @@ export async function getTransactionReceiptMined(txHash, interval) {
   } else {
     throw new Error("Invalid Type: " + txHash)
   }
+}
+
+export async function handleSetWallet(walletName, metamaskWalletAddress) {
+  if (walletName == 'metamask') {
```

Comment:
No, don't do this, you are already in the `metamask` directory so obviously the wallet is Metamask. What you need to do is to create a routes-like object in the `common/lib` directory, preferably named `wallet.js` (if the name is not used). This file will contain routes to the designated set wallet functions just in case we would want to expand to a different wallet in the near future.

You can reference an example I made here for the `debio-indexer`: https://github.com/debionetwork/debio-indexer/blob/9960abc9188a6ac6fa6b1dadaf17fe7260c6fef8/src/substrate/substrate.handler.ts#L31

A quick TL;DR **pseudocode** would look something like:
```
import { handleSetWallet } from './metamask' as metamaskHandleSetWallet;

const walletHandler = {
  metamask: metamaskHandleSetWallet,
};

export async function handleSetWallet(walletName, metamaskWalletAddress){
   await walletHandler[walletName](metamaskWalletAddress);
}
```

Also don't forget to remove the `walletName` parameter on Metamask.